lib/xcodeproj/project/object/native_target.rb

```diff
@@ -398,7 +398,7 @@ def symbol_type
         #         the files references of the source files that should be added
         #         to the target.
         #
-        # @param  [Hash{String=>String}] compiler_flags
+        # @param  [String] compiler_flags
```

Comment:
Perhaps we could allow instead to pass a hash of `PBXBuildFile#settings`, which would allow to define the header visibility. But then we would need to maintain a list of settings, which are valid for different types of build phases.